site/content/en/topics/assumptions.md

```diff
@@ -0,0 +1,27 @@
+---
+title: "Assumptions for Provisioning Tools"
+linkTitle: "Assumptions for Provisioning Tools"
+weight: 1
+type: docs
+description: >
+    Cloud provider assumptions on Azure resources that provisioning tools should take care.
+---
+
+Here is a list of Azure resource assumptions that are required for cloud provider Azure:
+
+* All Azure resources are under the same tenant.
+* All the virtual machines are part of either VirtualMachineScaleSet (VMSS) or AvailabilitySet (VMAS).
```

Comment:
This we need to rework. There is no Azure requirement for Virtual Machines to be in an availability set or scale set. Availability sets are one of the options for HA but not the only one. In regions that support them, spreading VMs on availability zones is a better option to ensure availability. Availability zones is a separate physical zone with its own power source, network, and cooling. On the other hand, an availability set only provides different fault domains, which are logical hardware groups. In the event that an entire datacenter building is affected, a zone provides better resiliency, if available.

https://docs.microsoft.com/en-us/azure/virtual-machines/availability 

In fact, CAPZ already does this today by spreading virtual machines on zones if available, only putting them in availability set if zones are not available. It works wit cloud provider today so we need to ensure it doesn't break in the future.

Reply:
related to https://github.com/kubernetes/kubernetes/issues/97375

Reply:
VMAS/VMSS are not conflicted with AZ, they are different levels of ensuring the HA. VMAS/VMSS are required because some of cloud provider functionality need concepts of "node pool"

Reply:
Availability sets and Availability zones are two options for availability, you wouldn't generally use both, it's one or the other. Some k8s provisioning tools like cluster api (or openshift) have different ways to define a "node pool" that are not synonymous with an availability set or a scale set, like a machine deployment. In CAPZ, a machine deployment will result in VMs being spread in all available zones and not placed in an availability set. A standalone VM without AS should be a supported use case (and it already is today).

cc @wking who commented on this in kubernetes/kubernetes#97375

Reply:
makes sense, thanks for the explanation. Moved this under enableMultipleStandardLoadBalancers.